homeassistant/setup.py

```diff
@@ -213,17 +213,20 @@ async def _async_process_dependencies(
         if dep not in hass.config.components
     }
 
-    after_dependencies_tasks: dict[str, asyncio.Future[bool]] = {}
     to_be_loaded = hass.data.get(DATA_SETUP_DONE, {})
     for dep in integration.after_dependencies:
         if (
             dep not in dependencies_tasks
             and dep in to_be_loaded
             and dep not in hass.config.components
         ):
-            after_dependencies_tasks[dep] = to_be_loaded[dep]
+            dependencies_tasks[dep] = setup_futures.get(dep) or create_eager_task(
+                async_setup_component(hass, dep, config),
+                name=f"setup {dep} as after dependency of {integration.domain}",
+                loop=hass.loop,
+            )
```

Comment:
Should we add a comment here explaining why we don't just wait for the task in `to_be_loaded` like we did before this PR?